packages/block-library/src/comment-template/edit.js

```diff
@@ -119,12 +124,52 @@ export default function CommentTemplateEdit( {
 	const blockProps = useBlockProps();
 
 	const [ activeComment, setActiveComment ] = useState();
-	const { commentOrder, commentsPerPage } = useSelect( ( select ) => {
+	const {
+		commentOrder,
+		commentsPerPage,
+		threadCommentsDepth,
+		threadComments,
+	} = useSelect( ( select ) => {
 		const { getSettings } = select( blockEditorStore );
 		return getSettings().__experimentalDiscussionSettings;
 	} );
 	const { rawComments, blocks } = useSelect(
 		( select ) => {
+			// Show placeholder comments if we don't have a postId context.
+			// The structure of rawComment as an empty object allows
+			// inner blocks to render the default placeholders.
+			if ( ! postId ) {
+				// If displaying threaded comments is disabled, we only show one comment
+				if ( ! threadComments ) {
+					return { rawComments: [ { id: null } ] };
+				}
+
+				// In case that `threadCommentsDepth` is falsy, we default to a somewhat
+				// arbitrary value of 3.
+				// In case that the value is set but larger than 3 we truncate it to 3.
+				const commentsDepth = Math.min( threadCommentsDepth || 3, 3 );
+
+				// We set a limit in order not to overload the editor of empty comments.
+				const defaultCommentsToShow =
+					perPage <= commentsDepth ? perPage : commentsDepth;
+
+				// This is a little hacky. We get a comment structure like:
+				// [
+				//   {id: -1, parent: 0},
+				//   {id: -2, parent: -1},
+				//   {id: -3, parent: -2}
+				//  ]
+				// The idea is that comments with a negative ID will not appear in a
+				// "real" system.
```

Comment:
🤯  😆

Reply:
It would be nice to extract all the logic that generates fake data and cover it with unit tests so it's easier to update if we ever have to. We already have `utils.js` file and some related tests.

Reply:
Another issue of this approach is that we try to fetch those negative id comments on the Rest API, having a 404 per placeholder.

![Screenshot 2022-01-31 at 11 28 01](https://user-images.githubusercontent.com/37012961/151777624-61428a60-311b-48ca-9361-772cfca43af2.png)

Also, it would be cool to tell the user that the threaded comments are replies, not everyone may know this when starting a FSE site.

Reply:
This is something we definitely want to avoid. It would flood server logs 😅